src/modules/register/Register.jsx

```diff
@@ -252,210 +240,124 @@ class RegisterForm extends React.Component {
-		} else {
-			status.password = '';
-			status.submit = '';
-		}
-		if (this.state.nameActive || this.state.emailActive) {
-			status.next = ' show-next';
-		} else {
-			status.next = '';
-		}
-		if (this.state.emailActive || this.state.passwordActive) {
-			status.back = ' show-back';
-		} else {
-			status.back = '';
-		}
+		const {active, showPassword} = this.state;
+
+		const inputProps = {
+			[this.SCREENS.NAME]: {
+				type: 'text',
+				name: "What's your name?",
+				autoComplete: 'given-name',
+				value: this.state.name
+			},
+			[this.SCREENS.EMAIL]: {
+				type: 'email',
+				name: "What's your email?",
+				autoComplete: 'email',
+				value: this.state.email
+			},
+			[this.SCREENS.PASSWORD]: {
+				type: showPassword ? 'text' : 'password',
+				name: 'Create your password',
+				autoComplete: 'current-password',
+				value: this.state.password,
+				children: (
+					<CheckBoxWrapper className="password-toggle">
+						<input className="eye-toggle" type="button" name="showPassword" id="showPassword" onClick={this.togglePassword} />
+					</CheckBoxWrapper>
+				)
+			}
+		};
 
 		return (
 			<Form onSubmit={this.submit}>
-				<div className="form-inner">
-					<Input
-						ref={this.inputs.name}
-						onChange={this.onChange('name')}
-						class={status.name}
-						type={'text'}
-						name={"What's your name?"}
-						autoComplete={'given-name'}
-					/>
-					<Input
-						ref={this.inputs.email}
-						onChange={this.onChange('email')}
-						class={status.email}
-						type={'email'}
-						name={"What's your email?"}
-						autoComplete={'email'}
-					/>
-
-					<div className={'input-container' + status.password}>
-						<input
-							ref={this.inputs.password}
-							type={this.state.showPassword ? 'text' : 'password'}
-							name={'password'}
-							id={'password'}
-							autoComplete={'current-password'}
-							value={this.state.password}
-							onChange={this.onChange('password')}
-						/>
-						<label htmlFor={'password'}>Create your password</label>
-
-						<CheckBoxWrapper className="password-toggle">
-							<input
-								className="eye-toggle"
-								type={'button'}
-								name={'showPassword'}
-								id={'showPassword'}
-								onClick={this.togglePassword}
-							/>
-						</CheckBoxWrapper>
-					</div>
-
-					<div className={'step' + status.back} onClick={this.goBack} />
-					<div className={'step' + status.next} onClick={this.showNext} />
-					<button className={status.submit} type={'submit'}>
-						Done
-					</button>
+				<div className="form-inner" onKeyPress={this.handleKeyPress}>
+					<Input onChange={this.onChange(active)} {...inputProps[active]} key={active} />
+					{this.getPrevScreen() && <div className="step show-back" onClick={this.goBack} />}
+					{this.getNextScreen() ? (
+						<div className="step show-next" onClick={this.showNext} />
+					) : (
+						<button className="show-submit" type="submit">
+							Done
+						</button>
+					)}
```

Comment:
In these lines you can see we're only rendering the buttons if they should be shown. This fixes a bug around the submit button firing too early if you press enter because there is a `<button type="submit">` on the page. And it lets us get rid of all the styling for managing `display`!